bolt-modules/boltlib/lib/puppet/functions/get_target.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+
+# Get a single target from inventory if it exists, otherwise create a new Target.
+#
+# **NOTE:** Calling `get_target` inside an `apply` block with a
+# version 2 inventory creates a new Target object.
+# `get_target('all')` returns an empty array.
+# **NOTE:** Only compatible with inventory v2
+Puppet::Functions.create_function(:get_target) do
+  # @param name A Target name.
+  # @return A single target, either new or from inventory.
+  # @example Create a new Target from a URI
+  #   get_target('winrm://host2:54321')
+  # @example Get an existing Target from inventory
+  #   get_target('existing-target')
+  dispatch :get_target do
+    param 'Boltlib::TargetSpec', :name
+    return_type 'Target'
+  end
+
+  def get_target(name)
+    inventory = Puppet.lookup(:bolt_inventory)
+    # Bolt executor not expected when invoked from apply block
+    executor = Puppet.lookup(:bolt_executor) { nil }
+    executor&.report_function_call(self.class.name)
+
+    unless inventory.version > 1
+      raise Puppet::ParseErrorWithIssue
+        .from_issue_and_stack(Bolt::PAL::Issues::UNSUPPORTED_INVENTORY_VERSION, action: 'get_target')
+    end
+
+    inventory.get_target(name)
```

Comment:
Is this not just a special case of `get_targets`? Should we just call that and get the first element?

Reply:
`get_targets` wont add a target to inventory. The desire for `get_target` called in a plan with a target name that does not exist in inventory to be part of the `all` group makes this a special case. 
Also unrelated, i noticed a pretty bad bug in this method. I invoke `existing_targets = @targets.key` before resolving the targets with `target_array = flat_target_list(target)` in the next line which can leave `@targets` un-populated with targets from inventory... Will push an update soon.

Reply:
Actually, I think I can use `get_targets` here in addition to the new `ensure_in_all_group` method which was implemented after this. Thanks.